sdk/storage/storage-blob/src/Pipeline.ts

```diff
@@ -2,234 +2,103 @@
-   *
-   * @type {KeepAliveOptions}
-   * @memberof StoragePipelineOptions
-   */
-  keepAliveOptions?: KeepAliveOptions;
-
-  /**
-   * Configures the HTTP pipeline logger.
-   *
-   * @type {IHttpPipelineLogger}
-   * @memberof StoragePipelineOptions
-   */
-  logger?: IHttpPipelineLogger;
-  /**
-   * Configures the HTTP client to send requests and receive responses.
-   *
-   * @type {IHttpClient}
-   * @memberof StoragePipelineOptions
-   */
-  httpClient?: IHttpClient;
 }
 
 /**
- * Creates a new Pipeline object with Credential provided.
+ * Creates a ServiceClientOptions object with Credential provided.
  *
  * @export
  * @param {SharedKeyCredential | AnonymousCredential | TokenCredential} credential Such as AnonymousCredential, SharedKeyCredential
  *                                                  or a TokenCredential from @azure/identity.
  * @param {StoragePipelineOptions} [pipelineOptions] Optional. Options.
- * @returns {Pipeline} A new Pipeline object.
+ * @returns {ServiceClientOptions} ServiceClientOptions for use with @azure/core-http
  */
 export function newPipeline(
   credential: SharedKeyCredential | AnonymousCredential | TokenCredential,
   pipelineOptions: StoragePipelineOptions = {}
-): Pipeline {
-  // Order is important. Closer to the API at the top & closer to the network at the bottom.
-  // The credential's policy factory must appear close to the wire so it can sign any
-  // changes made by other factories (like UniqueRequestIDPolicyFactory)
-  const factories: RequestPolicyFactory[] = [
-    tracingPolicy(),
-    new KeepAlivePolicyFactory(pipelineOptions.keepAliveOptions),
-    new TelemetryPolicyFactory(pipelineOptions.telemetry),
-    new UniqueRequestIDPolicyFactory(),
-    new BrowserPolicyFactory(),
-    deserializationPolicy(), // Default deserializationPolicy is provided by protocol layer
-    new RetryPolicyFactory(pipelineOptions.retryOptions),
-    logPolicy({
+): ServiceClientOptions {
+  const userAgentParts: string[] = [];
+  if (isNode) {
+    if (pipelineOptions.userAgentOptions && pipelineOptions.userAgentOptions.userAgentPrefix) {
+      userAgentParts.push(pipelineOptions.userAgentOptions.userAgentPrefix);
+    }
+    userAgentParts.push(`azsdk-js-storageblob/${SDK_VERSION}`);
+  }
+  const userAgentPrefix = userAgentParts.join(" ");
+
+  const fullPipelineOptions: InternalPipelineOptions = {
+    ...pipelineOptions,
+    userAgentOptions: {
+      userAgentPrefix
+    },
+    loggingOptions: {
       logger: logger.info,
       allowedHeaderNames: StorageBlobLoggingAllowedHeaderNames,
       allowedQueryParameters: StorageBlobLoggingAllowedQueryParameters
-    })
-  ];
+    },
+    updatePipelinePolicies: (requestPolicyFactories) => {
+      // Start with a clone
+      let result = requestPolicyFactories.slice(0);
+      // Currently, the only way to remove the retry policies provided by createPipelineFromOptions is
+      // by doing some careful math
```

Comment:
Concerning about this. What if requestPolicyFactories has changed without updating storage SDK logic?

Reply:
Yes definitely a brittle approach :-D